src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -101,9 +101,11 @@ def trace(self) -> JinjaTrace:
             target_slice_idx = self.find_slice_index(alt_id)
             slice_length = content_info if literal else len(str(content_info))
             self.move_to_slice(target_slice_idx, slice_length)
-        return JinjaTrace(
-            self.make_template(self.raw_str).render(), self.raw_sliced, self.sliced_file
-        )
+        # TRICKY: append_to_templated is a workaround for the dbt templater.
+        # When dbt uses Jinja to render a template, a trailing newline will be
+        # removed. This adds it back. The Jinja templater doesn't need this.
+        templated_str = self.make_template(self.raw_str).render() + append_to_templated
+        return JinjaTrace(templated_str, self.raw_sliced, self.sliced_file)
```

Comment:
Sorry not getting this. How is this code only returning that for dbt but not for Jinja?

Reply:
`append_to_templated` is an optional parameter, defaulting to `""`. See line 521 of the dbt templater. It passes `"\n"` if the input file contains trailing newlines. Now see line 365 of the jinja templater. It doesn't pass that parameter, so we get the default value (empty string).

Reply:
Again, can we make the comment clearer in this regards?